core/field_checkbox.js

```diff
@@ -127,11 +155,22 @@ Blockly.FieldCheckbox.prototype.initView = function() {
   this.textElement_.setAttribute('y', Blockly.FieldCheckbox.CHECK_Y_OFFSET);
   Blockly.utils.dom.addClass(this.textElement_, 'blocklyCheckbox');
 
-  var textNode = document.createTextNode(Blockly.FieldCheckbox.CHECK_CHAR);
-  this.textElement_.appendChild(textNode);
+  this.textContent_.nodeValue =
+      this.checkChar_ || Blockly.FieldCheckbox.CHECK_CHAR;
   this.textElement_.style.display = this.value_ ? 'block' : 'none';
 };
 
+/**
+ * Set the character used for the check mark.
+ * @param {string} character The character to use for the checkmark.
+ */
+Blockly.FieldCheckbox.prototype.setCheckCharacter = function(character) {
+  this.checkChar_ = character;
+  if (this.textContent_) {
+    this.textContent_.nodeValue = character;
```

Comment:
If you pass this null, it sets the nodeValue to null. but the desired outcome is to set it to CHECK_CHAR just like the above so I think you need ``character || Blockly.FieldCheckbox.CHECK_CHAR`` here

Reply:
Last unresolved comment, and then we're good to go!!

Reply:
Fixed and added test.